graphql-dgs-extended-scalars/src/main/kotlin/com/netflix/graphql/dgs/autoconfig/DgsExtendedScalarsAutoConfiguration.kt

```diff
@@ -33,7 +34,7 @@ import org.springframework.context.annotation.Configuration
     havingValue = "true",
     matchIfMissing = true
 )
-@Configuration(proxyBeanMethods = false)
+@AutoConfiguration
```

Comment:
Another notable change in functionality in 3.0.0-M5.

> With this release, support for registering auto-configurations in spring.factories has been removed in favor of the imports file.

> A new `@AutoConfiguration` annotation has been introduced. It should be used to annotate top-level auto-configuration classes that are listed in the new `META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports` file, replacing `@Configuration`.

https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0.0-M5-Release-Notes#auto-configuration-registration

Reply:
Under the hood, `@AutoConfiguration` is basically a more advanced `@Configuration` with `proxyBeanMethods` set to false.

https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/AutoConfiguration.java